src/ray/core_worker/transport/direct_actor_transport.cc

```diff
@@ -102,10 +96,8 @@ void CoreWorkerDirectActorTaskSubmitter::DisconnectActor(const ActorID &actor_id
     // If there are pending requests, treat the pending tasks as failed.
     auto pending_it = pending_requests_.find(actor_id);
     if (pending_it != pending_requests_.end()) {
-      auto head = pending_it->second.begin();
-      while (head != pending_it->second.end()) {
-        auto request = std::move(head->second);
-        head = pending_it->second.erase(head);
+      auto& requests = pending_it->second;
+      for (auto& request: requests) {
```

Comment:
Why don't we erase from the queue anymore? Couldn't we end up with duplicate tasks in the queue if the `TaskManager` later retries one of these pending tasks?